lib/angstrom.ml

```diff
@@ -462,8 +462,12 @@ let peek_char =
     else if more = Complete then
       succ input pos more None
     else
-      let succ' input' pos' more' =
-        succ input' pos' more' (Some (Input.get_char input' pos'))
+      let rec succ' input' pos' more' =
+        if pos < Input.length input then
+          succ input' pos' more' (Some (Input.get_char input' pos'))
+        else if more' = Complete then
+          succ input' pos' more' None
```

Comment:
I'm not as familiar with this portion of Angstrom.  Should this branch be a call to `fail'` to make it clear they're doing the same thing?

Reply:
This section of code actually looks similar to the checks that `prompt` is performing.